File: InternetScanner/languages/java/InternetScanner.java
package bizx.internetscanner;
import java.net.*; import java.util.*;
public final class InternetScanner {
 public static String classify(String s)throws Exception{ InetAddress a=InetAddress.getByName(s); return (a.isAnyLocalAddress()||a.isLoopbackAddress()||a.isLinkLocalAddress()||a.isSiteLocalAddress())?"local/intranet":"public"; }
 public static boolean authorized(String s, Set<String> allow)throws Exception { return !classify(s).equals("public")||allow.contains(s); }
}

File: InternetScanner/languages/kotlin/InternetScanner.kt
package bizx.internetscanner
import java.net.InetAddress
object InternetScanner {
 fun classify(ip:String):String { val a=InetAddress.getByName(ip); return if(a.isAnyLocalAddress||a.isLoopbackAddress||a.isLinkLocalAddress||a.isSiteLocalAddress) "local/intranet" else "public" }
 fun authorized(ip:String,allow:Set<String>)=classify(ip)!="public"||allow.contains(ip)
}
